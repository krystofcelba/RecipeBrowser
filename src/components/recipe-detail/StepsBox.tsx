import React from 'react';
import { StyleSheet, View, Text, StyleProp, ViewStyle } from 'react-native';

import { StylesConstants } from 'src/assets/constants';
import i18n from 'src/assets/i18n';

interface Props {
  style: StyleProp<ViewStyle>;
  steps: [string];
}

const StepsBox = ({ style, steps }: Props) => (
  <View style={[styles.container, style]}>
    <Text style={styles.heading}>{i18n.t('steps')}</Text>
    {steps.map((step, i) => (
      <View style={styles.stepTextContainer} key={i}>
        <Text style={styles.stepNum}>{`${1 + i}.`}</Text>
        <Text style={styles.stepText}>{step}</Text>
      </View>
    ))}
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
  },
  heading: {
    fontSize: StylesConstants.detailBoxHeadingFontSize,
    fontWeight: 'bold',
    fontFamily: StylesConstants.fontFamily,
  },
  stepTextContainer: { flexDirection: 'row' },
  stepNum: {
    fontFamily: StylesConstants.fontFamily,
    fontWeight: 'bold',
    paddingLeft: StylesConstants.defaultMargin / 2,
  },
  stepText: { fontFamily: StylesConstants.fontFamily, paddingLeft: StylesConstants.defaultMargin / 2 },
  seasoningsContainer: {
    flexDirection: 'column',
    marginTop: StylesConstants.defaultMargin / 2,
    marginRight: StylesConstants.defaultMargin,
  },
});

export default StepsBox;
