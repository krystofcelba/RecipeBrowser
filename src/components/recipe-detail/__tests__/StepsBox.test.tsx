import React from 'react';
import { shallow } from 'enzyme';
import StepsBox from '../StepsBox';

const mockedRecipes = require('../../../services/__mockData__/recipes.json');

describe('StepsBox component', () => {
  const mockedRecipe = mockedRecipes[0];
  it('renders as expected', () => {
    const wrapper = shallow(<StepsBox style={{}} steps={mockedRecipe.steps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
