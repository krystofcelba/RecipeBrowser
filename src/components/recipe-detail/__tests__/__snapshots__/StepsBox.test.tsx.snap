// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepsBox component renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "Avenir",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Steps
  </Text>
  <View
    key="0"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "fontWeight": "bold",
          "paddingLeft": 5,
        }
      }
    >
      1.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      Halve eggs and remove yolk.
    </Text>
  </View>
  <View
    key="1"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "fontWeight": "bold",
          "paddingLeft": 5,
        }
      }
    >
      2.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      Mix yolks with mustard, cottage cheese and half of dill.
    </Text>
  </View>
  <View
    key="2"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "fontWeight": "bold",
          "paddingLeft": 5,
        }
      }
    >
      3.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      Fill egg halves with yolk mixture and garnish with remaining dill.
    </Text>
  </View>
</View>
`;
